### PR TITLE
Vjogi/ravcom 444 upgrade core module

### PR DESCRIPTION
**Pull request by vijayjogi-dm**
Merging changes from apache calcite to dm-calcite

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/schema/impl/ListTransientTable.java [hunk 1]
```diff
@@ -84,20 +84,19 @@ public ListTransientTable(String name, RelDataType rowType) {
...
 
+  @SuppressWarnings("rawtypes")
   @Override public Collection getModifiableCollection() {
     return rows;
   }
 
   @Override public Enumerable<@Nullable Object[]> scan(DataContext root) {
-    // add the table into the schema, so that it is accessible by any potential operator
-    requireNonNull(root.getRootSchema(), "root.getRootSchema()")
-        .add(name, this);
```
Comment (rushikeshsapate-dm) [inline core/src/main/java/org/apache/calcite/schema/impl/ListTransientTable.java:94]: can we retain this piece of code?

File: core/src/main/java/org/apache/calcite/util/RelToSqlConverterUtil.java [hunk 2]
```diff
@@ -63,32 +63,6 @@ public static void unparseHiveTrim(
...
   }
 
-  /**
-   * This method will make regex pattern based on the TRIM flag.
-   *
-   * @param call     SqlCall contains the values that needs to be trimmed
-   * @param trimFlag It will contain the trimFlag either BOTH,LEADING or TRAILING
-   * @return It will return the regex pattern of the character to be trimmed.
-   */
-  public static SqlCharStringLiteral makeRegexNodeFromCall(SqlNode call, SqlLiteral trimFlag) {
```
Comment (rushikeshsapate-dm) [inline core/src/main/java/org/apache/calcite/util/RelToSqlConverterUtil.java:73]: can we retain this method?

File: core/src/main/java/org/apache/calcite/runtime/CalciteResource.java [hunk 3]
```diff
@@ -42,13 +42,19 @@ public interface CalciteResource {
...
+  ExInst<CalciteException> limitStartCountOrAllNotAllowed(String a0);
+
+  @BaseMessage("''OFFSET start LIMIT count'' is not allowed under the current SQL conformance level")
+  ExInst<CalciteException> offsetLimitNotAllowed();
+
   @BaseMessage("APPLY operator is not allowed under the current SQL conformance level")
   ExInst<CalciteException> applyNotAllowed();
 
-  @BaseMessage("JSON path expression must be specified after the JSON value expression")
-  ExInst<CalciteException> jsonPathMustBeSpecified();
```
Comment (rushikeshsapate-dm) [inline core/src/main/java/org/apache/calcite/runtime/CalciteResource.java:49]: can we retain jsonPathMustBeSpecified() ?

File: core/src/main/java/org/apache/calcite/schema/impl/ListTransientTable.java (hunk 1 above)
Comment (vijayjogi-dm) [reply core/src/main/java/org/apache/calcite/schema/impl/ListTransientTable.java:94]: Will do suggested changes in next PR

File: core/src/main/java/org/apache/calcite/util/RelToSqlConverterUtil.java (hunk 2 above)
Comment (vijayjogi-dm) [reply core/src/main/java/org/apache/calcite/util/RelToSqlConverterUtil.java:73]: Will do suggested changes in next PR

File: core/src/main/java/org/apache/calcite/runtime/CalciteResource.java (hunk 3 above)
Comment (vijayjogi-dm) [reply core/src/main/java/org/apache/calcite/runtime/CalciteResource.java:49]: Will do suggested changes in next PR

**Pull request merged**